tests/on_target/tests/test_power.py

```diff
@@ -0,0 +1,229 @@
+
+    badge_data = {
+        "label": "psm_current uA",
+        "message": f"{average}",
+        "schemaVersion": 1,
+        "color": f"{color}"
+    }
+
+    # Save the JSON data to a file
+    with open(badge_filename, 'w') as json_file:
+        json.dump(badge_data, json_file)
+
+    logger.info(f"Minimum average current saved to {badge_filename}")
+
+def save_measurement_data(samples):
+    # Generate timestamps for each sample assuming uniform sampling interval
+    timestamps = [round(i * SAMPLING_INTERVAL, 2) for i in range(len(samples))]
+
+    with open(CSV_FILE, 'w', newline='') as csvfile:
+        fieldnames = ['Time (s)', 'Current (uA)']
+        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+
+        writer.writeheader()
+        for t, current in zip(timestamps, samples):
+            writer.writerow({'Time (s)': t, 'Current (uA)': current})
+
+    logger.info(f"Measurement data saved to {CSV_FILE}")
+
+def generate_time_series_html(csv_file, date_column, value_column, output_file="time_series_plot.html"):
+    """
+    Generates an HTML file with an interactive time series plot from a CSV file.
+
+    Parameters:
+    - csv_file (str): Path to the CSV file containing the time series data.
+    - date_column (str): Name of the column containing date or time information.
+    - value_column (str): Name of the column containing the values to plot.
+    - output_file (str): Name of the output HTML file (default is "time_series_plot.html").
+
+    Returns:
+    - str: The path to the generated HTML file.
+    """
+    # Load the CSV file
+    df = pd.read_csv(csv_file, parse_dates=[date_column])
+
+    # Create an interactive Plotly line chart
+    fig = px.line(df, x=date_column, y=value_column, title='OOB Current Consumption Plot')
+
+    # Save the plot to an HTML file
+    fig.write_html(output_file)
+
+    logger.info(f"HTML file generated: {output_file}")
+    return output_file
+
+
+@pytest.fixture(scope="module")
+def ppk2():
+    '''
+    This fixture sets up ppk measurement tool.
+    '''
+    ppk2s_connected = PPK2_API.list_devices()
+    ppk2s_connected.sort()
+    if len(ppk2s_connected) == 2:
+        ppk2_port = ppk2s_connected[0]
+        print(f"Found PPK2 at {ppk2_port}")
+    elif len(ppk2s_connected) == 0:
+        pytest.skip("No ppk found")
+    else:
+        pytest.skip(f"Multiple PPks found")
+
+    ppk2_dev = PPK2_API(ppk2_port, timeout=1, write_timeout=1, exclusive=True)
+
+    # get modifier might fail, retry 15 times
+    for _ in range(15):
+        try:
+            ppk2_dev.get_modifiers()
+            break
+        except Exception as e:
+            logger.error(f"Failed to get modifiers: {e}")
+            time.sleep(5)
+    else:
+        pytest.skip("Failed to get ppk modifiers after 10 attempts")
+
+
+    ppk2_dev.set_source_voltage(3300)
+    ppk2_dev.use_ampere_meter()  # set ampere meter mode
+    ppk2_dev.toggle_DUT_power("ON")  # enable DUT power
+
+    time.sleep(10)
```

Comment:
I dont get uart if dut is not powered ON. That is also why I used a single fixture. Otherwise I could have also split ppk2 and uart into independent fixtures.
And I added the sleep to give uart time to show up, maybe can be removed